tests/modules/aclcheck.c

```diff
@@ -0,0 +1,172 @@
+#define REDISMODULE_EXPERIMENTAL_API
+
+#include "redismodule.h"
+#include <errno.h>
+#include <assert.h>
+
+/* A wrap for SET command with ACL check on the key. */
+int set_aclcheck_key(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc < 3) {
+        return RedisModule_WrongArity(ctx);
+    }
+
+    /* Check that the key can be accessed */
+    RedisModuleUserID * userid = RedisModule_CreateModuleUserID(ctx);
+    int ret = RedisModule_ACLCheckKeyPermissions(userid, argv[1]);
+    if (ret != 0) {
+        RedisModule_ReplyWithError(ctx, "DENIED KEY");
+        RedisModule_FreeModuleUserID(userid);
+        return REDISMODULE_OK;
+    }
+
+    RedisModuleCallReply *rep = RedisModule_Call(ctx, "SET", "v", argv + 1, argc - 1);
+    if (!rep) {
+        RedisModule_ReplyWithError(ctx, "NULL reply returned");
+    } else {
+        RedisModule_ReplyWithCallReply(ctx, rep);
+        RedisModule_FreeCallReply(rep);
+    }
+
+    RedisModule_FreeModuleUserID(userid);
+    return REDISMODULE_OK;
+}
+
+/* A dummy command that takes a key and do acl check with deleted user. */
+int test_aclcheck_key_user_deleted(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    if (argc != 2) {
+        return RedisModule_WrongArity(ctx);
+    }
+
+    RedisModuleUser *testuser = RedisModule_CreateModuleUser("testuser1");
+    RedisModule_SetModuleUserACL(testuser, "allcommands");
+    RedisModule_SetModuleUserACL(testuser, "allkeys");
+    RedisModule_SetModuleUserACL(testuser, "on");
+    RedisModule_AuthenticateClientWithUser(ctx, testuser, NULL, NULL, NULL);
+
+    /* Check that the key can be accessed */
+    RedisModuleUserID * testuserid = RedisModule_CreateModuleUserID(ctx);
+
+    /* authenticated back to "default" user (so once we free testuser1 we will not disconnected */
+    RedisModule_AuthenticateClientWithACLUser(ctx, "default", 7, NULL, NULL, NULL);
+    RedisModule_FreeModuleUser(testuser);
+
+    /* verify that using the UserID of deleted user, fails*/
+    int ret = RedisModule_ACLCheckKeyPermissions(testuserid, argv[1]);
```

Comment:
when did we delete the user?
did you intend to run some RM_Call to delete it here?

Reply:
RedisModule_FreeModuleUser